siteimprove/admin/js/siteimprove.js

```diff
@@ -121,6 +121,11 @@
       if( this.method === "domain" ){
         _si.push(['input', this.url, this.token, function() { console.log('Inputted new javascript overlay file'); } ]); 
       } else {
+        // 0 = overlay-latest.js
+        // 1 = overlay-v1.js
+        if(this.version == 0 && this.preview) {
+          _si.push(['registerPrepublishCallback', getDomCallback, this.token]);
```

Comment:
Shouldn't you call `registerPrepublishCallback` if not in preview - just with "null"?

Reply:
I don't agree or perhaps understand your intention. This condition basically combines version and preview condition statement, and we won't call registerPrepublishCallback at all unless we're on version-latest and in preview. There's no reason to call it with null or create more lines of code of conditions, when this just really does it. We can't just call with null here for this statement, because that method doesn't exist in v1, so it will crash Smallbox (even with null as argument)

Reply:
According to the documentation, you _can_ call `registerPrepublishCallback` with a `null` argument (for where the callback goes) - see here: https://github.com/Siteimprove/CMS-plugin-v2/blob/master/doc/Smallbox/si-cms.md.
Calling it with "null" will tell the iFrame that it should not show the button to start prepublish, so if you're in a situation where you're not in preview in WP, you have to call that.

Reply:
For trailing; Every page, from edit to preview, each is a new page load. What Morten says makes sense in the case that changing from edit to preview doesn't require one.